src/components/layout/nav/Navbar.jsx

```diff
@@ -53,6 +54,9 @@ function navbarIconsReducer(state, action) {
 }
 
 const Navbar = () => {
+  const [navbarClassName, setNavbarClassName] = useState("navbar");
```

Comment:
generally we try to avoid this practice. because using "magic strings" as values like this is very brittle. it's very easy to make a typo in a string or not

let's do the following
* rename `changeNavbarClassNameForSpecificComponents ` to `navBarClassForLocation`
* implementation stays the same, but instead of setting state, it RETURNS the new class name
* delete line 194

now, for the scrolling thing, i see that you still want to do something. however, like i said, if we want to capture state, we want to avoid random strings. instead, if you need state, you should express it using a boolean, in terms of what you want to say. for example, isScrollStateGreaterThan90, or something like that.




cc @emrerdem1 @OmarAlithawi @yamoonz